### PR TITLE
CP-2920 [API Device][Multidevice] Connected Devices - stress test - after rebase

### PR DESCRIPTION
**Pull request by robertmudi**
JIRA Reference: [CP-2920]

### :memo: Scenario
 Prerequisite #1: Mock of Kompakt Device, Latest OS, 1x Sim Card, Signal available, Battery 100%
 Prerequisite #2: PIN Code or Passcode is not set, Device is “unlocked”
 Prerequisite #3: Spec file: specs/select-device/select-device.ts
 Prerequisite #4: Page Object: Select Device Page - select-device.page.ts

1. Open App
2. Connect 1st Kompakt
3. Connect 2nd Kompakt
4. Wait until Select a device to continue modal is displayed Page Object: Select Device Page - select-device.page.ts
5. Verify Connected devices 
6. Connect 3rd Kompakt,Connect 4th Kompakt, Connect 5th Kompakt, Connect 6th Kompakt
7. Verify Connected devices 
8. Click 6th Kompakt
9. Mock OVERVIEW and OUTBOX
10. Verify Overview Page
11. Disconnect all Kompakt's
12. Welcome page should be displayed

[CP-2920]: https://appnroll.atlassian.net/browse/CP-2920?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    const deviceSelectDrawer = await drawerPage.deviceSelectDrawer
+    await expect(deviceSelectDrawer).toBeDisplayed()
+
+    const deviceImageOnDrawer = await drawerPage.deviceImageOnDrawer
+    await expect(deviceImageOnDrawer).toBeDisplayed()
+
+    const firstDeviceOnDrawer = await drawerPage.getDeviceOnDrawer(
+      firstSerialNumber
+    )
+    const secondDeviceOnDrawer = await drawerPage.getDeviceOnDrawer(
```
Comment (lkowalczyk87) [inline apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:77]: `secondDeviceOnDrawer` is unused

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    await firstDeviceOnDrawer.waitForClickable()
+    await firstDeviceOnDrawer.click()
+  })
+  it("Verify Overview Page for 1st device", async () => {
+    const kompaktImage = await OverviewKompaktPage.kompaktImage
+    await expect(kompaktImage).toBeDisplayed()
+    await expect(kompaktImage).toHaveAttribute("src", kompaktImageRegex)
+
+    const kompaktOsVersion = await OverviewKompaktPage.kompaktOsVersion
+    await expect(kompaktOsVersion).toBeDisplayed()
```
Comment (lkowalczyk87) [inline apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:90]: you may want to check what text contains `kompaktOsVersion` element

Comment (lkowalczyk87) [reply apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:90]: same for devices below

File: apps/mudita-center-e2e/src/helpers/add-device.helper.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { E2EMockClient } from "../../../../libs/e2e-mock/client/src"
+import { overviewDataWithOneSimCard } from "../../../../libs/e2e-mock/responses/src"
+
+export const addDeviceWithGivenSerialNumber = (serialNumber: string) => {
```
Comment (lkowalczyk87) [inline apps/mudita-center-e2e/src/helpers/add-device.helper.ts:4]: function is not in use, so please remove it

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    await expect(kompaktNetworkName).toHaveText("T-Mobile")
+
+    const kompaktBatteryLevelValue =
+      await OverviewKompaktPage.kompaktBatteryLevelValue
+    await expect(kompaktBatteryLevelValue).toBeDisplayed()
+    await expect(kompaktBatteryLevelValue).toHaveText("100%")
+  })
+
+  it("Switch to 2nd device", async () => {
+    const secondDeviceOnDrawer = await drawerPage.getDeviceOnDrawer(
```
Comment (lkowalczyk87) [inline apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:107]: I don't see the part where the drawer is opened. Am I missing something?

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts (hunk 1 above)
Comment (robertmudi) [reply apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:77]: fixed

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts (hunk 2 above)
Comment (robertmudi) [reply apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:90]: added await expect(kompaktOsVersion).toHaveText("Mudita OS") for each device

File: apps/mudita-center-e2e/src/helpers/add-device.helper.ts (hunk 3 above)
Comment (robertmudi) [reply apps/mudita-center-e2e/src/helpers/add-device.helper.ts:4]: removed

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts (hunk 4 above)
Comment (robertmudi) [reply apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:107]: it is opened while 2nd device is connected, then it clicks on the 2nd device from the top and opens it for further verification of the elements on the overview page

File: apps/mudita-center-e2e/src/specs/overview/e2e-mock-sample.e2e.ts [hunk 5]
```diff
@@ -74,6 +74,54 @@ describe("E2E mock sample - overview view", () => {
     await expect(drawerHeader).toBeDisplayed()
   })
 
+  it("Add third device", async () => {
```
Comment (tomaszmaleckimudita) [inline apps/mudita-center-e2e/src/specs/overview/e2e-mock-sample.e2e.ts:77]: This test fails. I think you should not edit whole example, if you want you can create a new one.

🔴  
```
[0-0] Error in "E2E mock sample - overview view.Remove first device"
Error: element ("//p[contains(text(), 'Offline')]") still not displayed after 6000ms
```

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+        serialNumber: device.serialNumber,
+      })
+
+      await browser.pause(6000)
+    }
+  })
+
+  it("Check Drawer Modal and Switch to 1st device", async () => {
+    const deviceSelectDrawer = await drawerPage.deviceSelectDrawer
+    await expect(deviceSelectDrawer).toBeDisplayed()
```
Comment (tomaszmaleckimudita) [inline apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:69]: I would call this test and change variables to something more fitting - this is Connected Devices Page / Modal - this is not a Drawer Page. How would next test where there will be an actual drawer ?


`Mudita Center.2.3.1-mockdev.45.app`

🔴  Test fails on first to phones 
```
Error: Expect $(`[data-testid="device-select-drawer-content"]`) to be displayed
Error: Expect $(`[data-testid="generic-view-image"]`) to be displayed
Error: Expect $(`//*[@data-testid="drawer-device-wrapper-KOM1234567892"]`) to be displayed
```

```
Expected: "displayed"
Received: "not displayed"
    at /Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:69:38
    at step (/Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:33:23)
    at Object.next (/Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:14:53)
    at fulfilled (/Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Comment (tomaszmaleckimudita) [reply apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:69]: Screenshot 2024-08-12 at 8 50 39 AM

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+        path: device.path,
+        serialNumber: device.serialNumber,
+      })
+
+      await browser.pause(6000)
+    }
+  })
+
+  it("Check Drawer Modal and Switch to 3rd device", async () => {
+    const deviceSelectDrawer = await drawerPage.deviceSelectDrawer
```
Comment (tomaszmaleckimudita) [inline apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:178]:
```
[0-0] Error in "Kompakt switching devices.Check Drawer Modal and Switch to 3rd device"
Error: Expect $(`[data-testid="device-select-drawer-content"]`) to be displayed

Expected: "displayed"
Received: "not displayed"
    at /Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:182:38
    at step (/Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:33:23)
    at Object.next (/Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:14:53)
    at fulfilled (/Users/tomasz/Dev/Mudita-Center/mudita-center/apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```
```
Error: Expect $(`[data-testid="generic-view-image"]`) to be displayed
Error: Expect $(`[data-testid="icon-DotsInBox"]`) to have text
```

**Review by tomaszmaleckimudita — CHANGES_REQUESTED** [review]

**Review by tomaszmaleckimudita — CHANGES_REQUESTED** [review]

File: apps/mudita-center-e2e/src/page-objects/select-device.page.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+import { index } from "@orama/orama/dist/components"
+import Page from "./page"
+
+class SelectDevicePage extends Page {
+  public get devicesList() {
+    return $("[data-testid='devices-list']")
+  }
+  public get availableDevice() {
```
[the remark this answers is not in the document]
Comment (robertmudi) [reply apps/mudita-center-e2e/src/page-objects/select-device.page.ts:13]: fixed and added asstertion in the tesT: 
 const availableDevices = selectDevicePage.availableDevices
    await expect(availableDevices).toBeDisplayed()

File: apps/mudita-center-e2e/src/page-objects/select-device.page.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    return $$('[data-testid="available-device"]')
+  }
+  public get selectDeviceSerialNumber() {
```
[the remark this answers is not in the document]
Comment (robertmudi) [reply apps/mudita-center-e2e/src/page-objects/select-device.page.ts:16]: I would leave it as plural as we get one SN per device, it is more readable

File: apps/mudita-center-e2e/src/page-objects/select-device.page.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    return $$('[data-testid="available-device-serial-number"]')
+  }
+  public get selectDeviceName() {
```
[the remark this answers is not in the document]
Comment (robertmudi) [reply apps/mudita-center-e2e/src/page-objects/select-device.page.ts:19]: I would leave it as plural as we get one name per device, it is more readable

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+      await selectDevicePage.getDeviceOnSelectModal(1)
+    const fourthDeviceOnSelectModal =
+      await selectDevicePage.getDeviceOnSelectModal(2)
+
+    await expect(thirdDeviceOnSelectModal).toBeDisplayed()
+    await expect(fourthDeviceOnSelectModal).toBeDisplayed()
+
+    const selectDeviceModalSerialNumbers =
+      selectDevicePage.selectDeviceSerialNumber
+    const selectDeviceModalNames = selectDevicePage.selectDeviceName
```
[the remark this answers is not in the document]
Comment (robertmudi) [reply apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:153]: I would leave it as plural as we call one name per device, it is more readable

File: apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+    const thirdDeviceOnSelectModal =
+      await selectDevicePage.getDeviceOnSelectModal(1)
+    const fourthDeviceOnSelectModal =
+      await selectDevicePage.getDeviceOnSelectModal(2)
+
+    await expect(thirdDeviceOnSelectModal).toBeDisplayed()
+    await expect(fourthDeviceOnSelectModal).toBeDisplayed()
+
+    const selectDeviceModalSerialNumbers =
+      selectDevicePage.selectDeviceSerialNumber
```
[the remark this answers is not in the document]
Comment (robertmudi) [reply apps/mudita-center-e2e/src/specs/stress-tests/connected-devices-stress-test.ts:152]: I would leave it as plural as we call one SN per device, it is more readable

**Review by tomaszmaleckimudita — APPROVED** [review]

**Pull request merged**